scipy/optimize/tests/test_trustregion_krylov.py

```diff
@@ -0,0 +1,103 @@
+"""
+Unit tests for Krylov space trust-region subproblem solver.
+
+To run it in its simplest form::
+  nosetests test_optimize.py
+
+"""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.optimize._trlib import (TRLIBQuadraticSubproblem)
+from numpy.testing import (TestCase, assert_, assert_array_equal,
+                           assert_almost_equal,
+                           assert_equal, assert_array_almost_equal,
+                           assert_array_less, run_module_suite)
+
+KrylovQP = TRLIBQuadraticSubproblem(tol_rel_i=1e-8, tol_rel_b=1e-6)
+
+class TestKrylovQuadraticSubproblem(TestCase):
+
+    def test_for_the_easy_case(self):
```

Comment:
I think it might worth including a few tests, to cover other special cases:
- The case interior convergence happens (``hits_boundary=false``)
- The case the gradient ``g`` is equal to zero (``norm(g)=0``)
- The case the gradient ``g`` is very close to zero but is not zero

By what I have seen you can reuse, without much modifications, many of the tests I have used in ``scipy/optimize/tests/test_trustregion_exact.py``, what can save you some time, since I have implemented there the tests I am suggesting.